maze_transformer/training/train_save_files.py

```diff
@@ -6,7 +6,7 @@
 from maze_transformer.training.config import ConfigHolder
 
 
-@freeze
```

Comment:
It looks like `freeze` can only act on objects, not types. It raises an exception when it recursively calls `freeze(TRAIN_SAVE_FILES.__dict__)`, since it can't act on a `mappingproxy`.